Hide Amazon SES keys in MailPoet dashboard - STOMAIL-7318

triple0t opened the pull request:
## Description

This PR enhances security in the MailPoet dashboard by implementing masked input fields for sensitive API credentials. The changes prevent accidental exposure of Amazon SES and SendGrid API keys while still allowing users to view credentials when needed through a show/hide toggle functionality.

**Why this change is needed:**
API keys and secrets displayed in plain text pose a security risk as they can be accidentally exposed during screenshots, screen sharing, or shoulder surfing. This is particularly important for Amazon SES access keys, secret keys, and SendGrid API keys which provide access to external email services.

**What was implemented:**
- Created a new reusable `PasswordInput` component with show/hide toggle functionality
- Applied the `PasswordInput` component to Amazon SES access key and secret key fields
- Applied the `PasswordInput` component to SendGrid API key field
- Refactored the existing premium key input to use the new `PasswordInput` component for consistency
- Added appropriate styling for the password toggle button

## Code review notes

The implementation follows these patterns:
- The `PasswordInput` component extends the existing `Input` component props
- Uses WordPress i18n for translatable "Show"/"Hide" button text
- Maintains the same UI/UX patterns as other form inputs in the settings
- CSS classes follow existing naming conventions (`mailpoet-password-input-toggle`)

Key files changed:
- `assets/js/src/common/form/input/password-input.tsx` - New reusable component
- `assets/js/src/settings/pages/send-with/other/amazon-ses-fields.tsx` - Updated to use PasswordInput
- `assets/js/src/settings/pages/send-with/other/sendgrid-fields.tsx` - Updated to use PasswordInput
- `assets/js/src/common/premium-key/key-input.tsx` - Refactored to use PasswordInput

## QA notes

**Testing scenarios:**
1. Navigate to Settings > Send With > Other and select Amazon SES
   - Verify access key and secret key fields show as masked password inputs
   - Verify clicking "Show" reveals the actual values
   - Verify clicking "Hide" masks the values again
2. Navigate to Settings > Send With > Other and select SendGrid
   - Verify API key field shows as masked password input with toggle functionality
3. Navigate to Settings > Advanced > Premium
   - Verify premium key input maintains show/hide functionality
4. Test form submission and data persistence works correctly with masked inputs
5. Verify existing functionality for saving and loading API keys is not affected

**Visual verification:**
- Password toggle buttons should be consistently styled across all forms
- Toggle state should be maintained per field independently
- Fields should default to masked state on page load


## Linked PRs

_N/A_

## Linked tickets

STOMAIL-7318

## After-merge notes

_N/A_

## Tasks

- [ ] I have added a changelog entry (pcNwfB-4Ov-p2#how-to-write-a-changelog)
- [ ] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [ ] I added sufficient test coverage
- [ ] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

## Preview

[Preview in WordPress Playground](https://account.mailpoet.com/playground/new/branch:stomail-7318-hide-amazon-ses-keys-in-mailpoet-dashboard)

_The latest successful build from `stomail-7318-hide-amazon-ses-keys-in-mailpoet-dashboard` will be used. If none is available, the link won't work._

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Introduced a password input field with visibility toggle functionality for improved user experience when entering sensitive information.

* **Enhancements**
  * Updated Amazon SES and SendGrid API key fields to use the new password input, ensuring keys are masked by default.
  * Improved styling for password input toggles.

* **Refactor**
  * Simplified premium key input by utilizing the new password input component.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

triple0t committed:
Add PasswordInput component with toggle functionality

- Introduced a new PasswordInput component that allows users to toggle password visibility.
- Updated styles for the password input toggle button.
- Exported the PasswordInput from the form index for easier access.

STOMAIL-7318

triple0t committed:
Use PasswordInput for amazon ses access key, secret key, and sendgrid api key

STOMAIL-7318

triple0t committed:
Refactor KeyInput component to use PasswordInput and remove toggle button functionality

- Replaced the existing input type with PasswordInput for better password handling.
- Removed the toggle button for revealing/hiding the premium key, simplifying the component.

STOMAIL-7318

triple0t committed:
Add changelog entry

STOMAIL-7318

triple0t committed:
Enhance PasswordInput component with accessibility improvements

- Added aria-labels for the toggle button to improve screen reader support.
- Simplified the return statement for better readability.

STOMAIL-7318

NeosinneR commented on mailpoet/assets/css/src/components-plugin/_forms.scss:
```diff
@@ -243,3 +243,8 @@ progress::-moz-progress-bar {
 }
+
+.mailpoet-password-input-toggle {
+  height: 34px;
```
The `Show/Hide` button overflows when used inside the small input components, and it's missing border radius (this one was present also on the `Key Activation` page).

<img width="1519" alt="Screenshot 2025-07-04 at 14 05 33" src="https://github.com/user-attachments/assets/94f3c225-2963-476c-bb29-0da88a0653f0" />

```suggestion
  border-radius: 0 3px 3px 0 !important;
  height: 100%;
```

<img width="907" alt="Screenshot 2025-07-04 at 14 08 00" src="https://github.com/user-attachments/assets/e8ee7f5e-6aef-4b29-ba0a-e678a0b6f025" />

NeosinneR reviewed (changes requested):
Nice improvement @triple0t, works as expected. 

I only see a small UI issue below.

triple0t committed:
Update mailpoet/assets/css/src/components-plugin/_forms.scss

Fix overflow and border radius issues for the PasswordInput component.

https://github.com/mailpoet/mailpoet/pull/6246#discussion_r2185205608

Co-authored-by: Ján Mikláš <neosinner@gmail.com>

Aschepikov reviewed (approved):
GJ!
